web/modules/custom/usagov_benefit_finder/modules/usagov_benefit_finder_content/src/Controller/CheckDataController.php

```diff
@@ -0,0 +1,593 @@
+<?php
+
+namespace Drupal\usagov_benefit_finder_content\Controller;
+
+/**
+ * Class CheckDataController
+ * @package Drupal\usagov_benefit_finder_content\Controller
+ */
+class CheckDataController {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The file system service.
+
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * Thee file repository service.
+   *
+   * @var \Drupal\file\FileRepositoryInterface
+   */
+  protected $fileRepository;
+
+  /**
+   * The file url generator service.
+   *
+   * @var \Drupal\Core\File\FileUrlGeneratorInterface
+   */
+  protected $fileUrlGenerator;
+
+  /**
+   * Active database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * Retrieves the currently active request object.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   */
+  protected $request;
+
+  /**
+   * The JSON data mode.
+   *
+   * @var string
+   */
+  protected $mode;
+
+  /**
+   * The langcode.
+   *
+   * @var string
+   */
+  protected $langcode;
+
+  /**
+   * The expanded.
+   *
+   * @var string
+   */
+  protected $expanded;
+
+  /**
+   * Constructs a new LifeEventController object.
+   */
+  public function __construct() {
+    $this->entityTypeManager = \Drupal::service('entity_type.manager');
+    $this->fileSystem = \Drupal::service('file_system');
+    $this->fileRepository = \Drupal::service('file.repository');
+    $this->fileUrlGenerator = \Drupal::service('file_url_generator');
+    $this->database = \Drupal::service('database');
+    $this->request = \Drupal::request();
+  }
+
+  /**
+   * Checks benefit finder data.
+   */
+  public function checkData() {
+    $this->mode = "draft";
+
+    // Get langcode.
+    if (empty($this->langcode)) {
+      $this->langcode = $this->request->get('langcode') ?? "en";
+    }
+
+    // Get expanded.
+    if (empty($this->expanded)) {
+      $this->expanded = $this->request->get('expanded') ?? "false";
+    }
+
+    $help = <<<EOD
+<h1>Benefit Finder Content Report</h1>
```

Comment:
I found this report at http://localhost/benefit-finder/content/report?langcode=en&expanded=true and ... it doesn't seem to work. Or perhaps more likely, I haven't put in the data it would report on.